File: Reqnroll/Chapter13/13.2-TheShareableScenarioContextPattern/Before/WIMP.ShareableScenarioContextSample.App/Services/AuthenticationService.cs
using WIMP.ShareableScenarioContextSample.App.Infrastructure;

namespace WIMP.ShareableScenarioContextSample.App.Services;

public static class AuthenticationService
{
    public static void Login(string customerName)
    {
        UserSession.Current.LoggedInCustomerName = customerName;
    }

    public static bool IsLoggedIn(string? customerName)
    {
        return customerName != null && UserSession.Current.LoggedInCustomerName == customerName;
    }
}

File: CucumberJVM/Chapter13/13.1-TheLocalScenarioContextPattern/Before/src/main/java/com/wimp/app/services/AuthenticationService.java
package com.wimp.app.services;

import com.wimp.app.infrastructure.UserSession;

public class AuthenticationService {
    public static void login(String customerName) {
        UserSession.getCurrent().setLoggedInCustomerName(customerName);
    }

    public static boolean isLoggedIn(String customerName) {
        return customerName.equals(UserSession.getCurrent().getLoggedInCustomerName());
    }
}
